src/NUnitFramework/framework/Attributes/TestFixtureSourceAttribute.cs

```diff
@@ -109,7 +108,7 @@ public IEnumerable<TestSuite> BuildFrom(Type type)
         {
             Type sourceType = SourceType ?? type;
 
-            foreach (TestFixtureParameters parms in GetParametersFor(sourceType))
+            foreach (ITestFixtureData parms in GetParametersFor(sourceType))
```

Comment:
Shouldn't we also remove the same unnecessary cast below? (https://github.com/nunit/nunit/pull/2971/files#diff-a7991f3c3308d3394ecebfc7b72ba252R128)

Reply:
@mikkelbu 
I guess that's why the test is failing. Will have a look.
Thanks!